src/main/java/com/tlswe/awsmock/ec2/control/MockEc2Controller.java

```diff
@@ -350,4 +356,46 @@ public void restoreAllMockEc2Instances(final Collection<AbstractMockEc2Instance>
         }
     }
 
+
+    /**
+     * Clean up terminated mock instances after a pre-defined period.
+     * Period is defined in aws-mock.properties (or if not overridden, as defined in aws-mock-default.properties)
+     *
+     * @param period
+     *            time period to clean up terminated instances
+     */
+    public void cleanupTerminatedInstances(final long period) {
+
+        TimerTask internalTimerTask = new TimerTask() {
+
+            /**
+             * this method is triggered every pre-defined period
+             */
+            private String terminatedState = AbstractMockEc2Instance.InstanceState.TERMINATED.getName();
+
+            @Override
+            public void run() {
```

Comment:
Runtime exceptions thrown here can cause the timer which schedules this task crash and fail to trigger any further occurrences. You may have to carefully handle exceptions in every TimerTask or alternatively use java.util.concurrent.ScheduledExecutorService instead of Timer.